search/search-client-connect/src/main/java/io/camunda/search/connect/jackson/CustomOffsetDateTimeDeserializer.java

```diff
@@ -31,14 +34,26 @@ public OffsetDateTime deserialize(final JsonParser parser, final Deserialization
 
     OffsetDateTime parsedDate;
     try {
+      // parse with the defined formatter, covering ~90% of all data reads
       parsedDate = OffsetDateTime.parse(parser.getText(), formatter);
-    } catch (final DateTimeParseException exception) {
-      parsedDate =
-          ZonedDateTime.parse(
-                  parser.getText(),
-                  DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss")
-                      .withZone(ZoneId.systemDefault()))
-              .toOffsetDateTime();
+    } catch (final DateTimeParseException dtpe1) {
+      try {
+        // try with the default colon-based formatter, full seconds offset and 'Z' allowed
+        // this can be introduced by data writers that don't use the ConnectConfiguration
+        parsedDate = OffsetDateTime.parse(parser.getText());
```

Comment:
@tmetzke I am wondering if this is already covered by `yyyy-MM-dd'T'HH:mm:ss.SSSXXXX` ?

Reply:
No, the `XXXX` (4x) offset pattern is only for "no colon, full second details allowed". It cannot parse colon-based offset. See also [DateTimeFormatter docs](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/time/format/DateTimeFormatter.html):
> Offset X and x: This formats the offset based on the number of pattern letters. One letter outputs just the hour, such as '+01', unless the minute is non-zero in which case the minute is also output, such as '+0130'. Two letters outputs the hour and minute, without a colon, such as '+0130'. Three letters outputs the hour and minute, with a colon, such as '+01:30'. **Four letters outputs the hour and minute and optional second, without a colon, such as '+013015'**. Five letters outputs the hour and minute and optional second, with a colon, such as '+01:30:15'. Six or more letters throws IllegalArgumentException. Pattern letter 'X' (upper case) will output 'Z' when the offset to be output would be zero, whereas pattern letter 'x' (lower case) will output '+00', '+0000', or '+00:00'.

The parsing works with similar semantics, allowing **up to** seconds-detail in offsets, but less is also fine. But colons it cannot detect.

Reply:
Yes, OffsetDateTime uses `XXXXX` (5). `yyyy-MM-dd'T'HH:mm:ss.SSS[XXXXX][XXXX]` could probably parse both.

Reply:
I tried with that notation in local tests but this doesn't work for me. Maybe I am doing something wrong though 😄

Reply:
Yeah, I didn't test it too much. It fixed the IT tests locally for me.

Reply:
Yes, these changes work too!